src/coreComponents/mesh/generators/VTKMeshGenerator.hpp

```diff
@@ -0,0 +1,280 @@
+   * of the attribute index and the type of the element.
+   * For instance, the cells of a mesh with two regions will hold the attribute "1", or "2". The CellBlocks will
+   * be instantiated according to the attribute and the type of the cells. If the region "1" has wedges, tetrahedron
+   * and hexahedron, three CellBlocks will be created names 1_tetrahedron, 1_wedges and 1_hexahedron.
+   * The ElementRegions have to be be defined in the XML file.
+   *
+   * The pointsets of surface are defined in the same way, using the same property names "attribute" defined in the 
+   * input mesh. The pointsets will hold a name that is just the attribute index. For instance, if a mesh has three
+   * surfaces of interest, with triangles and/or quads holding an attribute value of 1, 2 or 3, three pointsets named
+   * "1", "2" and "3" will be instantiated by this method
+   * @param[in] domain the DomainPartition to be written
+   */
+  virtual void generateMesh( DomainPartition & domain ) override;
+
+protected:
+
+  /**
+   * @brief This function provides capability to post process input values prior to
+   * any other initialization operations.
+   */
+  void postProcessInput() override final;
+
+private:
+
+  /**
+   * @brief Return a VTK controller for multiprocessing.
+   */
+  vtkSmartPointer<vtkMultiProcessController> GetVTKController();
+
+  /**
+   * @brief Load the VTK file into the VTK data structure
+   */
+  vtkSmartPointer<vtkUnstructuredGrid> LoadVTKMesh();
+
+  /**
+   * @brief Redistribute the mesh among the available MPI ranks
+   * @details this method will also generate global ids for points and cells in the VTK Mesh
+   * @param[in] loadedMesh the mesh that was loaded on one or several MPI ranks
+   */
+  void RedistributeMesh(vtkUnstructuredGrid * loadedMesh);
+
+  /**
+   * @brief Copy the VTK mesh nodes into the nodeManager of GEOSX
+   * @param[in] nodeManager the NodeManager of the domain in which the poiints will be copied.
+   * @return the global length of the mesh
+   */
+  double WriteMeshNodes(NodeManager & nodeManager);
+
+  /**
+   * @brief Compute the potential rank neighbor list
+   * @details Fills the metisNeighbor list in \p domain.
+   * @param[in] domain the DomainPartition in which the neighbhor list will be computed
+   */
+  void ComputePotentialNeighborLists( DomainPartition & domain);
+
+  /** 
+   * @brief Get the attribute data array from the VTK mesh
+   * @return a pointer to the vtkIntArray containing the attributes if it exists, nullptr otherwise.
+   */
+  vtkIntArray * GetAttributeDataArray();
+
+  /** 
+   * @brief Get the cell global id data array from the VTK mesh
+   * @return a pointer to the vtkIdTypeArray containing the global ids
+   */
+  vtkIdTypeArray * GetCellGlobalIdDataArray();
+
+  /**
+   * @brief This methos is used to preprocess the the VTK mesh and count the number of cells, facets, regions
+   * and surfaces.
```

Comment:
Does it count over the ranks, the whole process?

Reply:
it count independently of every rank